addons/mail/static/src/widgets/discuss/discuss.js

```diff
@@ -31,14 +31,21 @@ export const DiscussWidget = AbstractAction.extend({
 
         this._lastPushStateActiveThread = null;
         this.env = Component.env;
-        Component.env.services.messaging.modelManager.messagingCreatedPromise.then(() => {
+        Component.env.services.messaging.modelManager.messagingCreatedPromise.then(async () => {
             const messaging = Component.env.services.messaging.modelManager.messaging;
             const initActiveId = this.options.active_id ||
                 (this.action.context && this.action.context.active_id) ||
                 (this.action.params && this.action.params.default_active_id) ||
                 'mail.box_inbox';
             this.discuss = messaging.discuss;
-            this.discuss.update({ initActiveId });
+            this.discuss.update({ initActiveId, isOpen: true });
+            await messaging.initializedPromise;
+            if (!this.discuss.wasInitThreadConsidered) {
+                this.discuss.update({ wasInitThreadConsidered: true });
+                if (!this.discuss.thread) {
+                    this.discuss.openInitThread();
+                }
+            }
```

Comment:
1st LOC it good and relevant.
The rest is off.

The opening of init thread in discuss should all be modelling with (computed) fields.

I suppose `await messaging.initializedPromise;` is to do proper opening of thread after insert of channel data of `init_messaging`. This is unchanged behavior from LOC in `messaging_initializer.js` before change... but would be much nicer if all this was computed field to open the thread.

It looks like `wasInitThreadConsidered` is kind of a semaphore to avoid doing modelling changes from action more than once. Again all this would be more robust and more natural to read if all this was (computed) fields.